File: duck-typing.py
class Duck :
    def cry(self):
        return "quack-quack"
class Chick :
    def cry(self):
        return "wing-chicken"
    
def howDoesACry(animal):
    animalName = animal.__class__.__name__
    sound = animal.cry()
    print("{}은 {}".format(animalName,sound))

howDoesACry(Duck())
howDoesACry(Chick())

File: duck-typing.js
class Duck{
    cry(){
        return 'quack-quack'
    }
}

class Chick{
    cry(){
        return 'wing-chicken';
    }
}

function howDoesACry(animal){
    console.log(animal.constructor.name + '은 ' + animal.cry())
}

howDoesACry(new Duck())
howDoesACry(new Chick())